presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestOptimizedPartitionedOutputOperatorQueries.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestOptimizedPartitionedOutputOperatorQueries
```

Comment:
This class doesn't really test optimized partitioned output operators (it was tested in `TestOptimizedPartitionedOutputOperatorQueries`).

So maybe just call it `AbstractTestExchangeQueries`? Maybe also rename the subclass names?

Reply:
Makes sense. However "exchangeQueries" is likely to be confused with "exchangeClient". Since this is the sending side(repartitioning) not the receiving side (exchange) how about renaming it to AbstractTestRepartitionQueries?

Reply:
@yingsu00 : `AbstractTestRepartitionQueries` sounds good 😃 . Or `AbstractTestQueriesWithRepartition`? Both look good to me.